### PR TITLE
Rate limiting tests vsr

### PR DESCRIPTION
**Pull request by vepatel**
### Proposed changes
- Add vsr test scenarios for rate-limiting functionality,
- Add test profiling in edge pipeline.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: tests/suite/test_rl_policies_vsr.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+            f"{req_url}{v_s_route_setup.route_m.paths[0]}",
+            headers={"host": v_s_route_setup.vs_host},
+        )
+        print(resp.status_code)
+        delete_policy(kube_apis.custom_objects, pol_name, v_s_route_setup.route_m.namespace)
+        self.restore_default_vsr(kube_apis, v_s_route_setup)
+        assert resp.status_code == 500
+
+    @pytest.mark.parametrize("src", [rl_vsr_override_vs_spec_src, rl_vsr_override_vs_route_src])
+    def test_overide_vs_vsr(
```
Comment (Dean-Coakley) [inline tests/suite/test_rl_policies_vsr.py:277]:
```suggestion
    def test_override_vs_vsr(
```

**Review by Dean-Coakley — APPROVED** [review]
:shipit:

**Review by LorcanMcVeigh — APPROVED** [review]
